vnext/ReactUWP/Modules/AlertModuleUwp.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "AlertModuleUwp.h"
+#include <winrt/Windows.UI.Xaml.Controls.h>
+
+#include <Utils\ValueUtils.h>
+
+#pragma warning(push)
+#pragma warning(disable : 4146)
+#include <cxxreact/JsArgumentHelpers.h>
+#pragma warning(pop)
+namespace winrt {
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+} // namespace winrt
+
+#if _MSC_VER <= 1913
+// VC 19 (2015-2017.6) cannot optimize co_await/cppwinrt usage
+#pragma optimize("", off)
+#endif
+
+namespace react {
+namespace uwp {
+//
+// Alert
+//
+
+const char *AlertModule::name = "RCTAlert";
+
+AlertModule::~AlertModule() = default;
+
+std::string AlertModule::getName() {
+  return name;
+}
+
+std::map<std::string, folly::dynamic> AlertModule::getConstants() {
+  return {};
+}
+
+std::vector<facebook::xplat::module::CxxModule::Method>
+AlertModule::getMethods() {
+  return {
+      Method(
+          "showAlert",
+          [](folly::dynamic args, Callback actionResult) {
+            AlertModule::showAlert(args, actionResult);
+          }),
+
+  };
+}
+
+/*static*/ winrt::fire_and_forget AlertModule::showAlert(folly::dynamic &args, Callback actionResult) {
+
+  std::wstring alertTitle;
+  std::wstring alertContent;
+  std::wstring alertPBT;
+  std::wstring alertSBT;
+  std::wstring alertCBT;
+
+  for (auto arr : args) {
+  
+  for (auto &pair : arr.items()) {
+    const std::string &propertyName = pair.first.getString();
+    const folly::dynamic &propertyValue = pair.second;
+        
+      if (propertyName == "title") {
+        alertTitle = react::uwp::asHstring(propertyValue);
+      }
+        
+      if (propertyName == "message") {
+        alertContent = react::uwp::asHstring(propertyValue);
+      }
+    
+      if (propertyName == "buttonPositive") {
+        alertPBT = react::uwp::asHstring(propertyValue);
+      }
+      if (propertyName == "buttonNegative") {
+        alertSBT = react::uwp::asHstring(propertyValue);
+      }
+      if (propertyName == "buttonNeutral") {
+        alertCBT = react::uwp::asHstring(propertyValue);
+      }
```

Comment:
If it'll only be one of these at a time you could consider using else if's here to make it slightly more efficient (it won't check the rest of the statements once it finds the right one). To me, it also currently implies that a propertyName could meet multiple of these conditions.